ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/State/Infra.hs

```diff
@@ -170,56 +170,84 @@ reconstructSummary (History.Shape shape) transition (HardForkState st) =
     go :: Exactly xs' EraParams
        -> Telescope (K Past) (Current f) xs'
        -> NonEmpty xs' EraSummary
+    go ExactlyNil t = case t of {}
     go (ExactlyCons params ss) (TS (K Past{..}) t) =
         NonEmptyCons (EraSummary pastStart (EraEnd pastEnd) params) $ go ss t
-    go (ExactlyCons params ExactlyNil) (TZ Current{..}) =
```

Comment:
This `go (ExactlyCons params ExactlyNil) (TZ Current{..})` case is the buggy one. Just because it's the last era the code knows about doesn't mean there won't be another era. My intuition is that it should instead be treated just like the `go (ExactlyCons params (ExactlyCons nextParams _)) (TZ Current{..})` case below.

By removing this case, we will be changing many kinds of downstream behaviors: eg local clients can send the `GetInterpreter` query (aka `QueryEraHistory` in `cardano-node` repo) in order to be able to do their own slot<->time translations. Because of this bug, such requests while in the last known era currently have no time horizon (they'll happily translate slot<->time a million years in the future), but as of the bugfix they response to that query (and everything else downstream of `reconstructSummary`) will have the intended finite horizon (mostly "one stability window").

In particular, it's possible some SPO has written their own client and is unfortunately depending on this buggy unlimited translation range during Alonzo.

Reply:
cc @kevinhammond